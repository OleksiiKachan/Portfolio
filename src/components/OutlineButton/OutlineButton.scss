@import '../../assets/style/variables';

.outlineButton {
  border: 2px solid $color-gold;
  background: transparent;
  padding: 10px 25px;
  color: $color-white;
  font-weight: 300;
  cursor: pointer;

  &:hover {
    background: $color-gold;
    font-weight: 500;
    color: #fff;
  }
}
