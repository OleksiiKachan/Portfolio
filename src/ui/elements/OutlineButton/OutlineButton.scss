@import '../../assets/style/variables';

.outlineButton {
  border: 2px solid;
  background: transparent;
  padding: 10px 25px;
  font-weight: 300;
  cursor: pointer;
  transition: all 200ms ease-in-out;

  &_gold {
    border-color: $color-gold;
    color: $color-white;

    &:hover {
      background: $color-gold;
      color: $color-surface;
    }
  }

  &_white {
    border-color: $color-surface;
    color: $color-surface;

    &:hover {
      border-color: $color-surface;
      background: $color-surface;
      color: $color-black;
      font-weight: 400;
    }
  }

  &_dark {
    border-color: $color-grey;
    color: $color-grey;

    &:hover {
      background: $color-grey;
      color: $color-surface;
    }
  }

  &:hover {
    font-weight: 500;
  }
}
