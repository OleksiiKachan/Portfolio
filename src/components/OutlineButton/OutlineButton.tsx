import React, { Component, ReactNode } from 'react';
import classNames from 'classnames';
import './OutlineButton.scss';

/*
    <OutlineButton onClick=() className=''/>
*/
type PropsType = {
  onClick: () => void;
  children: ReactNode;
  className?: string;
};

export default ({ onClick, children, className }: PropsType) => {
  return (
    <button
      type="button"
      onClick={onClick}
      className={classNames('outlineButton', className)}
    >
      {children}
    </button>
  );
};
