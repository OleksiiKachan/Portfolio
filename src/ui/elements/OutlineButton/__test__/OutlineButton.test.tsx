import React from 'react';
import { shallow } from 'enzyme';
import OutlineButton from '../OutlineButton';

describe('OutlineButton', () => {
  it('Gold button', () => {
    const component = shallow(
      <OutlineButton type="gold" onClick={() => {}}>
        Caption
      </OutlineButton>
    );

    expect(component).toMatchSnapshot();
  });
  it('Dark button', () => {
    const component = shallow(
      <OutlineButton type="dark" onClick={() => {}}>
        Caption
      </OutlineButton>
    );

    expect(component).toMatchSnapshot();
  });
});
