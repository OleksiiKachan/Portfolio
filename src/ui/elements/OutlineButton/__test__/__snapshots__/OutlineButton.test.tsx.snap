// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlineButton Dark button 1`] = `
<button
  className="outlineButton outlineButton_dark"
  onClick={[Function]}
  type="button"
>
  Caption
</button>
`;

exports[`OutlineButton Gold button 1`] = `
<button
  className="outlineButton outlineButton_gold"
  onClick={[Function]}
  type="button"
>
  Caption
</button>
`;
